src/sql/azdata.proposed.d.ts

```diff
@@ -976,4 +976,168 @@ declare module 'azdata' {
 			expiresOn?: number
 		}
 	}
+
+	export enum DataProviderType {
+		TableDesignerProvider = 'TableDesignerProvider'
+	}
+
+	export namespace dataprotocol {
+		export function registerTableDesignerProvider(provider: designers.TableDesignerProvider): vscode.Disposable;
+	}
+
+	export namespace designers {
+
+		export function openTableDesigner(providerId: string, tableInfo: TableInfo): Promise<void>;
+
+		export interface TableDesignerProvider extends DataProvider {
+			getTableDesignerInfo(table: TableInfo): Promise<TableDesignerInfo>;
+			processTableEdit(table: TableInfo, data: DesignerData, edit: DesignerEdit): Promise<DesignerEditResult>;
+		}
+
+		/**
+		 * The information of the table.
+		 */
+		export interface TableInfo {
+			server: string;
```

Comment:
should we reference a ConnectionInfo object or something like that for the connection details?